### DISCUSSION TITLE
changing Neper mesh

### DISCUSSION THREAD

**Discussion by Mahdieh-karami** [neperfepx/neper#922] [Q&A] [answered]
Hi, 
I wanted to know if it is possible to change mesh style in Neper. For example I want each grain has smooth boundary like in the picture  . 
[](![Capture](https://github.com/user-attachments/assets/689186e2-5bbc-452e-a031-7e62c65eb1bf))
![Capture](https://github.com/user-attachments/assets/9b112c02-0f40-4156-9927-860a7d1cdf20)

**Comment by rquey** [comment, marked as answer]
Use tet meshing:

```
neper -T -n 10 -reg 1 -dim 2
neper -M n10-id1.tess
neper -V n10-id1.tess,n10-id1.msh -dataelt2dedgerad 0.002 -print img
```
![img](https://github.com/user-attachments/assets/6d576ecd-ea93-4467-91bf-287f51078d68)

Quad meshing has been discussed [here](https://github.com/neperfepx/neper/discussions/98).

**Reply by Mahdieh-karami** [reply]
Thanks for your response. I should use hex mesh . So there is no way to have smooth boundary with hex ?

**Reply by rquey** [reply]
The way is described in this discussion: https://github.com/neperfepx/neper/discussions/98, and more specifically this comment: https://github.com/neperfepx/neper/discussions/98#discussioncomment-318210.

**Reply by Mahdieh-karami** [reply]
Thanks.

**Reply by Mahdieh-karami** [reply]
Hi @rquey , Is it possible to do this for 3D as well ?

**Reply by rquey** [reply]
As far as I remember, yes, but you'll have to investigate...

**Reply by Mahdieh-karami** [reply]
Thanks.
